packages/gatsby/src/utils/page-data.ts

```diff
@@ -168,14 +168,12 @@ export async function flush(parentSpan?: Span): Promise<void> {
   const { pagePaths } = pendingPageDataWrites
   let writePageDataActivity
 
-  let nodeManifestPagePathMap
+  // we process node manifests in this location because we need to add the manifestId to the page data.
+  // We use this manifestId to determine if the page data is up to date when routing. Here we create a map of "pagePath": "manifestId" while processing and writing node manifest files.
+  // We only do this when there are pending page-data writes because otherwise we could flush pending createNodeManifest calls before page-data.json files are written. Which means those page-data files wouldn't have the corresponding manifest id's written to them.
+  const nodeManifestPagePathMap = await processNodeManifests()
```

Comment:
Why move this out of the if? We only use it when pagePaths.size > 0 so I guess putting it back in the if is us doing less work?

Reply:
I'm going to revert this and let that customer know they need to add a GraphQL query for it to work

Reply:
Thanks for pointing this out @wardpeet !

Reply:
:shrug: I just noticed it, I have no idea what processNodeManifests does

Reply:
Well your question made me re-read the code comment and avoid a bug so thank you haha